frontend/src/javascript/utils/intervalsUtils.js

```diff
@@ -1,5 +1,50 @@
-export function areIntervalsIntersect(aStart, aEnd, bStart, bEnd) {
-    return aEnd > bStart && aStart < bEnd;
+export function areIntervalsIntersectOrTouching(aStart, aEnd, bStart, bEnd) {
+    return aEnd >= bStart && aStart <= bEnd;
+}
+
+export function applyChangeOnInterval(aStart, aEnd, change) {
+    let result = { start: aStart, end: aEnd };
+
+    if (change.end <= aStart) {
```

Comment:
Przeglądajac te ify musiałem poświęcić chwilkę (krótką, bo te warunki są proste i da się zrozumieć o co w nich chodzi), ale jak się tą chwilkę przemnoży przez ilość tych ifów to już jest dłuższa chwilka. Jak tą dłuższą chwilkę przemnożysz przez ilość osób przeglądających ten kod (a chciałbym zaznaczyć, że nasza firma, może w przyszłości liczyć nawet kilkaset pracowników) no to się robi długa chwila. Można to rozwiązać poprzez dodanie komentarza do każdego ifa jaki przypadek on rozwiązuje. W książce "Czysty kod" autor wielokrotnie podkreślał, że komentarze, to jest tylko łatka na nieczytelny kod. Jeżeli kod jest nieczytelny to dużo lepiej przerobić go na czytelny kod, niż dodawać komentarze (osobiście zgadzam się z tą opinią). Dlatego proponuję podobnie jak to było z funkcjami `isSubinterval` dodać dla każdego takiego warunku osobną funkję - jej nazwa będzie jednocześnie komentarzem co dany if obsługuje. Moim zdaniem kod typu:
`if (isChangeBeforeInterval) {...}`
`if (isChangeAfterInterval) {...}`
`if (isWholeIntevalChanged) {...}`
będzie bardziej czytelny. (chciałem nawet pójść krok dalej i tak samo przerobić ciała tych ifów, jednak tutaj wydaje mi się, że ciężko by było nadać jakieś sensowne nazwy takim funkcjom).

Być może uznasz, że jest to nadmiarowy zabieg i że ta "długa chwila" poświęcona na czytanie tego pliku przez x osób, będzie i tak krótsza niż czas poświęcony przez Ciebie na refaktoryzację tego, ale wydaje mi się, że to jest dobra praktyka i warto ją stosować. Cytat z wspomnianej ksiązki: "W rzeczywistości stosunek czasu spędzanego na czytaniu do czasu spędzanego na pisaniu jest jak 10:1.". Dlatego jestem taki drobiazgowy i "czepiam się" szczegółów :D.